### ISSUE TITLE
attach_file and attach_file_mini break on osx

### ISSUE THREAD

**Issue by magnatelee** [StanfordLegion/realm#13]
I removed attach_file from travis temporarily because it has been breaking frequently on OS X: https://github.com/StanfordLegion/legion/commit/8400aeed2286256ab8a8fe702ca5a15309fd4773.
Put this test back once the issue is resolved.

I was able to reproduce this issue quite reliably, although it doesn't seem to reproduce on Sean's Mac. Here is a command to reproduce the issue:

CC_FLAGS="-std=c++98 -DLEGION_SPY" DEBUG=1 make  -j;
while [ $? -eq 0 ]; do LEGION_FREEZE_ON_ERROR=1 ./attach_file -logfile out_%.log; done

**Comment by elliottslaughter** [comment]
Yes, we still have this turned off in CI:

https://gitlab.com/StanfordLegion/legion/-/blob/master/test.py?ref_type=heads#L155-159

So my first inclination would be to make a MR with turns it back on (removing the conditional) and see if it fails.

**Comment by elliottslaughter** [comment]
MR to re-enable these tests at: https://gitlab.com/StanfordLegion/legion/-/merge_requests/1826

We'll see if it passes or fails.

**Comment by elliottslaughter** [comment]
CI appears to be happy so I'm closing this now.

**Comment by muraj** [comment]
> As an aside, it would be nice if someone wrote a comprehensive Realm-only test for this functionality, so we don't need to rely on Legion-level tests to ensure this continues to work.

Agreed, I'll file an issue for this.  I actually have an integration test for this while working on cufile implementation.  I will need to revive that.
